app/schema.json

```diff
@@ -42,11 +42,92 @@
         "initialYDomain": { "$ref": "#/definitions/domain" },
         "autocompleteSource": { "type": "string" },
         "genomePositionSearchBoxVisible": { "type": "boolean" },
-        "genomePositionSearchBox": { "type": "object", "TODO": "more precise" },
+        "genomePositionSearchBox": { "$ref": "#/definitions/genomePositionSearchBox" },
         "chromInfoPath": { "type": "string" },
-        "tracks": { "type": "object", "TODO": "more precise" },
-        "layout": { "type": "object", "TODO": "more precise" },
-        "overlays": { "type": "array" }
+        "tracks": { "$ref": "#/definitions/tracks_object" },
+        "layout": { "$ref": "#/definitions/layout" },
+        "overlays": { "type": "array", "items": { "$ref": "#/definitions/overlay" } }
+      }
+    },
+
+    "genomePositionSearchBox": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["autocompleteServer", "autocompleteId", "chromInfoServer", "chromInfoId", "visible"],
```

Comment:
autoCompleteServer and autoCompleteId shouldn't be required. You can have a Genome Position Search Box with just chromsome info.

Reply:
Just took them out of the required list. Thanks.